engine/apps/public_api/serializers/integrations.py

```diff
@@ -166,30 +169,155 @@ def validate_templates(self, templates):
+            for attr, template in template_from_request.items():
                 try:
-                    template = templates[backend_id.lower()][field]
+                    validated_data[PUBLIC_TEMPLATES_FIELDS[template_backend_name][attr]] = template
                 except KeyError:
-                    continue
+                    print(2)
+                    raise BadRequest(detail="Invalid template data")
+        elif template_from_request is None:
+            # if it's we receive None, it's needed to set template to default value
+            try:
+                template_to_set_to_default = PUBLIC_TEMPLATES_FIELDS[template_backend_name]
+                if type(template_to_set_to_default) is str:
+                    # if we receive None for plain template just set it to None
+                    validated_data[PUBLIC_TEMPLATES_FIELDS[template_backend_name]] = None
+                elif type(template_to_set_to_default) is dict:
+                    # if we receive None for nested template set all it's fields to None
+                    for key in template_to_set_to_default.keys():
+                        validated_data[PUBLIC_TEMPLATES_FIELDS[template_backend_name][key]] = None
+            except KeyError:
+                print(4)
+                raise BadRequest(detail="Invalid template data")
+        return validated_data
+
+    def _correct_validated_data_for_messaging_backends_templates(self, validated_data):
+        """
+        _correct_validated_data_for_messaging_backends_templates reshapes validated data to store them.
+        It converts data from "templates" dict to messaging_backends_templates field format.
+        Example:
+            {
+              "msteams": {
+                "title": Hello
+              }
+            }
+            Will be converted to
+
+            messaging_backends={"MSTEAMS": {"title": "Hello"},
+        """
+        templates_data_from_request = validated_data.get("templates", {})
+
+        messaging_backends_templates = self.instance.messaging_backends_templates if self.instance else {}
+
+        for backend_id, backend in get_messaging_backends():
+            backend_template = {}
+            if backend_id.lower() in templates_data_from_request:  # check to modify only templates from request data
+                template_from_request = templates_data_from_request[backend_id.lower()]
+            else:
+                continue
+            if template_from_request is None:
+                # If we receive None backend template, like {"msteams": None }, set all template fields to none.
+                for field in backend.template_fields:
+                    backend_template[field] = None
+            elif type(template_from_request) is dict:
+                # go through existing backend_template and update with values from request
+                backend_template = messaging_backends_templates.get(backend_id, {})
```

Comment:
```suggestion
                backend_template = messaging_backends_templates.get(backend.backend_slug, {})
```

Reply:
it's funny, but in db we are storing templates by backend_id, not slug)